numpy/random/mtrand.pyx

```diff
@@ -882,6 +882,14 @@ cdef class RandomState:
         array(['pooh', 'pooh', 'pooh', 'Christopher', 'piglet'], # random
               dtype='<U11')
 
+        Array-like value is internally converted to a ndarray so if you
+        want to preserve original values, convert it yourself first using
+        dtype object:
```

Comment:
nit:
```suggestion
        ``object`` dtype:
```

although this behavior spans across numpy, it may be more pronounced for this function due to selection of elements from the object. may probably help to call it out.